src/arithmetic/arithmetic_expression.c

```diff
@@ -1037,96 +1037,57 @@ SIValue AR_GT(SIValue *argv, int argc) {
 	SIValue a = argv[0];
 	SIValue b = argv[1];
 
-	if(SIValue_IsNull(a) || SIValue_IsNull(b)) return SI_NullVal();
-
-	assert(SI_TYPE(a) == SI_TYPE(b));
-
-	switch(SI_TYPE(a)) {
-	case T_STRING:
-		return SI_BoolVal(SIValue_Compare(a, b) > 0);
-	case T_INT64:
-	case T_DOUBLE:
-		return SI_BoolVal(SI_GET_NUMERIC(a) > SI_GET_NUMERIC(b));
-	default:
-		assert(false);
-	}
+	// TODO: Return type mismatch error.
+	if(!IS_COMPERABLE(a) || !IS_COMPERABLE(b)) assert(false);
+	int res = SIValue_Compare(a, b);
+	if(res == COMPARED_NULL) return SI_NullVal();
+	return SI_BoolVal(res > 0);
 }
 
 SIValue AR_GE(SIValue *argv, int argc) {
 	assert(argc == 2);
 	SIValue a = argv[0];
 	SIValue b = argv[1];
 
-	if(SIValue_IsNull(a) || SIValue_IsNull(b)) return SI_NullVal();
-
-	assert(SI_TYPE(a) == SI_TYPE(b));
-	//Type mismatch: expected Float, Integer, Point, String, Date, Time, LocalTime, LocalDateTime or DateTime
-	// but was Node (line 1, column 22 (offset: 21)) "match (n),(m) return n > m" ^
-
-	switch(SI_TYPE(a)) {
-	case T_STRING:
-		return SI_BoolVal(SIValue_Compare(a, b) >= 0);
-	case T_INT64:
-	case T_DOUBLE:
-		return SI_BoolVal(SI_GET_NUMERIC(a) >= SI_GET_NUMERIC(b));
-	default:
-		assert(false);
-	}
+	// TODO: Return type mismatch error.
+	if(!IS_COMPERABLE(a) || !IS_COMPERABLE(b)) assert(false);
```

Comment:
This doesn't accomplish the same thing as
https://github.com/RedisGraph/RedisGraph/pull/613/files#diff-f7fdc0d4bc00f4ef287d8d0392e8c60eR55

It determines whether each input is of a type that's capable of comparison, but not whether the two inputs can be compared with each other.